include/envoy/server/worker.h

```diff
@@ -35,6 +35,20 @@ class Worker {
   virtual void stop() PURE;
 
   /**
+   * Remove a listener from the worker.
```

Comment:
Might be useful to point out that as part of a normal removal flow we expect to `stopListener()` first to drain. I'm picturing a sequencing in which we first drain, wait a predefined amount of time, then remove. Could we have a completion for `stopListener()` that lets us know when no more connections  exist (e.g. they were all cleared by the H2 GOAWAY) so that we can complete the listener upgrade faster? 

The reason I think this might be useful is that when we update LDS configuration, it might be useful to know when all traffic is using the new LDS config (e.g. if a route table was embedded in the LDS config before/after). To know this, we either have to wait for the drain timeout (let's say it's a minute) or for all connections to close (which might happen in a few seconds).

It's more complex to do this, so just throwing it out there for discussion/consideration.